types/webpack-dev-server/index.d.ts

```diff
@@ -270,8 +274,20 @@ declare namespace WebpackDevServer {
          * This option allows us either to choose the current web-socket server
          * or to provide custom web-socket server implementation.
          */
-        webSocketServer?: boolean | 'sockjs' | 'ws' | string | (() => void) | Record<string, any>;
+        webSocketServer?:
+            | string
+            | boolean
+            | WebSocketType
+            | { type?: string | boolean | WebSocketType; options?: Record<string, any> };
+
+        /**
+         * Allows to set server and options (by default 'http')."
+         */
+        server?: string | ServerType | { type?: ServerType; options?: https.ServerOptions };
     }
+
+    type ServerType = 'http' | 'https' | 'spdy';
+    type WebSocketType = 'sockjs' | 'ws';
```

Comment:
For consistency, please define these new types above, before `Configuration` interface.